mlir/unittests/Target/LLVM/SerializeToLLVMBitcode.cpp

```diff
@@ -74,3 +98,52 @@ TEST_F(MLIRTargetLLVM, SKIP_WITHOUT_NATIVE(SerializeToLLVMBitcode)) {
   // Check that it has a function named `foo`.
   ASSERT_TRUE((*llvmModule)->getFunction("foo") != nullptr);
 }
+
+std::optional<SmallVector<char, 0>>
+TargetAttrImpl::serializeToObject(Attribute attribute, Operation *module,
+                                  const gpu::TargetOptions &options) const {
+  module->setAttr("serialize_attr", UnitAttr::get(module->getContext()));
+  std::string targetTriple = llvm::sys::getProcessTriple();
+  LLVM::ModuleToObject serializer(*module, targetTriple, "", "");
+  return serializer.run();
+}
+
+Attribute
+TargetAttrImpl::createObject(Attribute attribute, Operation *module,
+                             const SmallVector<char, 0> &object,
+                             const gpu::TargetOptions &options) const {
+  return gpu::ObjectAttr::get(
+      module->getContext(), attribute, gpu::CompilationTarget::Offload,
+      StringAttr::get(module->getContext(),
+                      StringRef(object.data(), object.size())),
+      module->getAttrDictionary());
+}
```

Comment:
Can you also add more documentation to the tests, out of the context of this PR it wouldn't be straightforward to understand the intent of all this I believe.

Reply:
Ok, wrt docs everywhere, would you prefer a fresh patch, or can I bundle the changes into this patch https://github.com/llvm/llvm-project/pull/95292 ? (they are related)

Reply:
`mlir/include/mlir/Dialect/GPU/IR/CompilationAttrInterfaces.td` isn't touched at all in https://github.com/llvm/llvm-project/pull/95292 ? I wouldn't add it to the changeset there.

Reply:
You're right, I'm touching `CompilationAttrs.td`. I'll merge that other one, create a patch with the docs and then revisit if we can change Operation* to GPUModuleOp.